annyang.js

```diff
@@ -3,8 +3,19 @@
 //! author  : Tal Ater @TalAter
 //! license : MIT
 //! https://www.TalAter.com/annyang/
-
-(function(undefined) {
+(function (root, factory) {
+  // istanbul ignore next
+  // jshint strict: false
+  if (typeof define === 'function' && define.amd) { // AMD
+    define([], function () {
+      return (root.annyang = factory(root, undefined));
+    });
+  } else if (typeof module === 'object' && module.exports) { // CommonJS
+    module.exports = factory(root, undefined);
+  } else { // Browser globals
+    root.annyang = factory(root, undefined);
+  }
+}(typeof window !== 'undefined' ? window : this, function (root, undefined) {
```

Comment:
Why do you need to check for the existence of window here?
I am guessing this comes in handy if using AMD or CommonJS in the browser/electron, but I would like to understand the code completely.